src/main.c

```diff
@@ -1227,10 +1272,67 @@ static void handle_get_component_line(void *data, const char *line) {
   }
 }
 
+static char* get_get_launch_components_cmd(char* sharedenv) {
+  const char basecmd[] = "%s %s/bin/zwe internal get-launch-components --config \"%s\" --ha-instance %s";
+  int size = strlen(zl_context.root_dir) + strlen(zl_context.config_path) + strlen(zl_context.ha_instance_id) + strlen(sharedenv) + sizeof(basecmd) + 1;
+  char *command = malloc(size);
+
+  snprintf(command, size, basecmd,
+           sharedenv, zl_context.root_dir, zl_context.config_path, zl_context.ha_instance_id);
+  
+  return command;
+}
+
+/**
+ * @brief Get the sharedenv. The function contemplates enclosing in quotes the values of the variables.
+ * 
+ * @return char* string representation of the shared_uss_env variable, e.g. VAR1="sample" VAR2=12345
+ */
+static char* get_sharedenv(void) {
+  char *output = NULL;
+  char *aux = NULL;
+
+  int required = 0;
+  for (char **env = shared_uss_env + 1; *env != 0; env++) { // First element is NULL, reserved to _BPX_SHAREAS
+    char *thisEnv = *env;
+    required += (strlen(thisEnv) + 3); // space + quotes
+  }
+
+  required++;
+  output = malloc(required);
+  aux = malloc(required);
+  for (char **env = shared_uss_env + 1; *env != 0; env++) { // First element is NULL, reserved to _BPX_SHAREAS
+    char *thisEnv = *env;
+    strcat(aux, thisEnv);
+    char *envName = strtok(aux, "=");
+    if (envName) {
+      strcat(output, envName);
+      char *envValue = &thisEnv[strlen(envName) + 1];
+      if (*envValue == '"') { // Env value is already enclosed in quotes
+        strcat(output, "=");
+        strcat(output, envValue);
+        trimRight(output, strlen(output));
+        strcat(output, " ");
+      } else {
+        strcat(output, "=\"");
+        strcat(output, envValue);
```

Comment:
Quetos are not escaped

Reply:
I'm not sure what you mean here, this code is from the previous version, just moved it up

Reply:
The JSON content is solved by `jsonToString`. The potential issue could come from `shared_uss_env[idx++] = thisEnv;`.

The proper solution there should probably be like this: 
```c
if (*thisEnv != '\"') {
 newstr = malloc(?);
 newstr[0] = 0; 
 strcat(newstr, key);
 strcat(newstr, "=");
 strcat(newstr, escapeString(thisEnv + length);
 length = newstr;
}
```

...anyway, I guess it is a bug, that could be solved in an issue. The workaround could be to use a different form in STC. And from another hand, it is pretty rear to set especially in this way string value with a quote.

Reply:
opened an issue based on testing done with these scenarios https://github.com/zowe/launcher/issues/76